jupyterlab_latex/__init__.py

```diff
@@ -225,6 +228,146 @@ def get(self, path = ''):
+        """Builds tuple that will be used to call the synctex view shell command.
+
+        Parameters
+        ----------
+        tex_name: string
+            This is the base name of the tex file, without the extension
+
+        pos: dict
+            A dictionary containing the position in the tex file
+            document to map.
+
+        returns:
+            A dictionary containing the mapped position data.
+
+        """
+        c = LatexConfig(config=self.config)
+
+        cmd = (
+            c.synctex_command,
+            'view',
+            '-i',
+            f'{pos["line"]}:{pos["column"]}:{tex_name+".tex"}',
+            '-o',
+            f'{tex_name+".pdf"}'
+            )
+
+        return cmd
+
+
+    @gen.coroutine
+    def run_synctex(self, cmd):
+        """Run commands sequentially, returning a 500 code on an error.
+
+        Parameters
+        ----------
+        command_sequence : list of tuples of strings
+            This is a sequence of tuples of strings to be passed to
+            `tornado.process.Subprocess`, which are to be run sequentially.
+            On Windows, `tornado.process.Subprocess` is unavailable, so
+            we use the synchronous `subprocess.run`.
+
+        Returns
+        -------
+        string
+            Response is either a success or an error string.
+
+        Notes
+        -----
+        - SyncTex processes only print to stdout, so errors are gathered from
+          there.
+
+        """
+        code, output = yield run_command(cmd)
+        if code != 0:
+            self.set_status(500)
+            self.log.error((f'SyncTex command `{" ".join(cmd)}` '
+                              f'errored with code: {code}'))
+        return output
+
+
+    @web.authenticated
+    @gen.coroutine
+    def get(self, path = ''):
+        """
+        Given a path, run SyncTex, and respond when done.
+        """
+        # Get access to the notebook config object
+        full_file_path = os.path.join(self.notebook_dir, path.strip('/'))
+        workdir = os.path.dirname(full_file_path)
+        base_name, ext = os.path.splitext(os.path.basename(full_file_path))
+
+        if not os.path.exists(full_file_path):
+            self.set_status(403)
+            out = f"Request cannot be completed; no file at `{full_file_path}`."
+        elif not os.path.exists(os.path.join(workdir, base_name + '.synctex.gz')):
+            self.set_status(403)
+            out = f"Request cannot be completed; no SyncTeX file found in `{workdir}`."
+        elif ext != '.tex' and ext != '.pdf':
+            self.set_status(400)
+            out = (f"The file `{ext}` does not end with .tex of .pdf. "
+                    "You can only run SyncTex on a file ending with .tex or .pdf.")
+        else:
+            if ext == '.pdf':
+                pos = {
+                    'page': self.get_query_argument('page', default='1'),
+                    'x': self.get_query_argument('x', default='0'),
```

Comment:
Can you comment on the dimensionality of these?

Reply:
What do you mean "dimensionality"?

Reply:
what is the dimension of the x and y values on the python side of things.

Reply:
Added a comment